### PR TITLE
T6219: Add support for container sysctl parameter

### PR DESCRIPTION
**Pull request by nvollmar**
I can't append to #3288, so I took the liberty to build upon it.
Aligned configuration with system sysctl and added completion with allowed sysctl values for container

## Change Summary
Allow setting sysctl settings for containers

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
https://vyos.dev/T6219

## Related PR(s)
#3288

## Component(s) name
container

## Proposed changes
Adds support for configuring a container with the sysctl option:
```
container {
    name c1 {
        allow-host-networks
        image "busybox:stable"
        sysctl {
            parameter kernel.msgmax {
                value "8192"
            }
        }
    }
}
```

## How to test
```
set container name c1 image busybox:stable
set container name c1 allow-host-networks
set container name c1 sysctl parameter kernel.msgmax value 8192
```

## Smoketest result
```
test_basic (__main__.TestContainer.test_basic) ... ok
test_dual_stack_network (__main__.TestContainer.test_dual_stack_network) ... 
IP address "192.0.2.1" can not be used for a container, reserved for the
container engine!

ok
test_ipv4_network (__main__.TestContainer.test_ipv4_network) ... 
IP address "192.0.2.1" can not be used for a container, reserved for the
container engine!

ok
test_ipv6_network (__main__.TestContainer.test_ipv6_network) ... 
IP address "2001:db8::1" can not be used for a container, reserved for
the container engine!

ok
test_uid_gid (__main__.TestContainer.test_uid_gid) ... 
Cannot set "gid" without "uid" for container

ok

----------------------------------------------------------------------
Ran 5 tests in 146.650s
```

## Checklist:
- [X] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [X] I have linked this PR to one or more Phabricator Task(s)
- [X] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [X] My commit headlines contain a valid Task id
- [X] My change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/container.xml.in [hunk 1]
```diff
@@ -71,6 +71,36 @@
               <multi/>
             </properties>
           </leafNode>
+          <node name="sysctl">
+            <properties>
+              <help>Configure namespaced kernel parameters of the container</help>
+              <priority>318</priority>
```
Comment (sever-sever) [inline interface-definitions/container.xml.in]: the container has priority 450
```
vyos@r4# /usr/libexec/vyos/priority.py | grep contain
       450  container.py                        ['container']

```
Do we really need priority here?

Comment (nvollmar) [reply interface-definitions/container.xml.in]: I don't know this functionality of the config definition. I accidentally copied that over from system sysctl.
I can remove it.

Comment (sever-sever) [reply interface-definitions/container.xml.in]: The priority option shows which component will start/apply first during load config/commit
For example, if we have a tunnel and configuration address for eth interface, firstly, we should assign the address to the ethX interfaces, and then you can use it in the tunnel.
```
vyos@r4# /usr/libexec/vyos/priority.py | match "eth|tun"
       300  interfaces_virtual-ethernet.py      ['interfaces', 'virtual-ethernet']
       318  interfaces_ethernet.py              ['interfaces', 'ethernet']
       321  interfaces_pseudo-ethernet.py       ['interfaces', 'pseudo-ethernet']
       380  interfaces_tunnel.py                ['interfaces', 'tunnel']
[edit]
vyos@r4# 
```
Delete the priority here, we do not need it in this case.

Comment (nvollmar) [reply interface-definitions/container.xml.in]: thanks, for the explanation, removed it.

File: src/completion/list_container_sysctl_parameters.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+#
+# Copyright (C) 2021 VyOS maintainers and contributors
```
Comment (sever-sever) [inline src/completion/list_container_sysctl_parameters.sh]: 2024

Comment (nvollmar) [reply src/completion/list_container_sysctl_parameters.sh]: fixed

**Review by sever-sever — APPROVED** [review]

**Comment by sever-sever** [comment]
Probably will be conflict with https://github.com/vyos/vyos-1x/pull/3607

**Comment by nvollmar** [comment]
Yep, just added that as separate commit to get the builds green.
Will rebase/remove once #3607 is merged

**Review by dmbaturin — APPROVED** [review]

**Comment by nvollmar** [comment]
rebased

File: smoketest/scripts/cli/test_container.py [hunk 3]
```diff
@@ -80,6 +80,7 @@ def test_basic(self):
 
         self.cli_set(base_path + ['name', cont_name, 'image', cont_image])
         self.cli_set(base_path + ['name', cont_name, 'allow-host-networks'])
+        self.cli_set(base_path + ['name', cont_name, 'sysctl', 'parameter', 'kernel.msgmax', 'value', '8192'])
```
Comment (c-po) [inline smoketest/scripts/cli/test_container.py]: Please not only set this value, but also validate it's properly set inside the container.

**Review by c-po — CHANGES_REQUESTED** [review]

File: smoketest/scripts/cli/test_container.py (hunk 3 above)
Comment (nvollmar) [reply smoketest/scripts/cli/test_container.py]: added a check

**Review by c-po — APPROVED** [review]

**Comment by nvollmar** [comment]
@sever-sever  would be great if that could be backported to Sagitta

**Comment by c-po** [comment]
@MergifyIo backport sagitta

**Comment by sever-sever** [comment]
@MergifyIo backport circinus-stream sagitta-stream

**Pull request merged**